src/behavior.ts

```diff
@@ -19,17 +22,18 @@ import { Stream } from "./stream";
 export type SemanticBehavior<A> = (time: Time) => A;
 
 @monad
-export abstract class Behavior<A> extends Reactive<A>
-  implements Observer<A>, Monad<A> {
+export abstract class Behavior<A> extends Reactive<A, BListener>
+  implements Parent<BListener>, Monad<A> {
   // Push behaviors cache their last value in `last`.
   // Pull behaviors do not use `last`.
   last: A;
-  nrOfListeners: number;
-  // The streams and behaviors that this behavior depends upon
-  dependencies: Cons<Reactive<any>>;
+  children: DoubleLinkedList<BListener> = new DoubleLinkedList();
   // Amount of nodes that wants to pull the behavior without actively
+  nrOfListeners: number;
   // listening for updates
   nrOfPullers: number;
+  pulledAt: number;
+  changedAt: number;
```

Comment:
We use `undefined` to represent that a behavior has never been updated so the type for `changedAt` should be `number | undefined`?